### DISCUSSION TITLE
I can't return to the dashboard in UserCrudController in the setupUpdateOperation function

### DISCUSSION THREAD

**Discussion by denison** [Laravel-Backpack/community-forum#572] [General] [answered]
Hi guys,

I have a validation on the user edit page. If true,  I want to return to dashaboard. 

I tested the 3 options in the image, and it doesn't return to the dashboard, the backpack returns to the same edit page.

image -> https://prnt.sc/cXMY2CtjocvW

someone who can help me ?

**Comment by christmex** [comment]
Hi @denison 

Try this, in your UserCrudController
```php
public function edit($id){
        $user = backpack_user(); //or anything this is just example
        // return parent::edit($id); // uncomment this for show edit user form
        if(!is_null($user)){
            if($user->hasRole('Super')){
                \Alert::success('something!')->flash();
                return redirect(backpack_url('dashboard'));
            }
        }
        
    }
```

this method overwrites the edit method from backpack updateOperation , idk if its legal to do that or not, if you want to return the edit page, you can use the ``` return parent::edit($id); ``` 

let me know if you found any issues, I also have the same question as you in my question haha, I'm still asking the correct way, but for now, you can use that way.

**Reply by christmex** [reply]
@denison here's another reference @tabacitu just answered my question, I thought we have the same question here https://github.com/Laravel-Backpack/community-forum/discussions/571

**Reply by denison** [reply]
Hi @christmex , 

putting the edit in UserCrudUser worked if the user does not have permission, does not access and returns to the dashboard. However, in case the user has access, it is giving an error.

Error -> https://prnt.sc/ky5BLjtuN_qd

code-> https://prnt.sc/9V9TqeV8Pvya

the error now is when I want to return the form.

**Comment by christmex** [comment, marked as answer]
Hi @denison 
based on what i see from your controller, did you extend your UserCrudController to UserCrudController from backpack?

try to extend your class with backpack UserCrudController, please take a look
```php
<?php

namespace App\Http\Controllers\Admin\Auth;
use \Backpack\PermissionManager\app\Http\Controllers\UserCrudController as BackpackUserCrudController; //add this line, remove this comment

class UserCrudController extends BackpackUserCrudController //extend this, remove this comment
{
// all of your code here
}
```
make sure you already extend the UserCrudController from backpack

then add this line inside your ``` register() ``` function placed at ```app/Providers/AppServiceProvider.php ```
```php
/**
     * Register any application services.
     */
    public function register(): void
    {
        $this->app->bind(
            \Backpack\PermissionManager\app\Http\Controllers\UserCrudController::class, //this is package controller
            \App\Http\Controllers\Admin\Auth\UserCrudController::class //this should be your own controller
        );
    }
```

let me know if you find any issues.

Thanks

**Reply by denison** [reply]
Thanks man .. its gone!
